src/ReferenceRepository/DataObjectReferenceRepository.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\ReferenceRepository;
+
+/**
+ * based on \Doctrine\Common\DataFixtures\ReferenceRepository.
+ *
+ * @template T of object
```

Comment:
I don't think the templating on this class work as they're now. As far as I know, `T` expands to *exactly one* type, which means you will not be able to store, e.g., the references to the DataObjects `X` *and* `Y` in this class (at least according to PhpStan).

If you want I can try to get it right.

Reply:
Please feel free to fix the templating annotations.